ur_controllers/src/gpio_controller.cpp

```diff
@@ -474,9 +474,9 @@ bool GPIOController::setPayload(const ur_msgs::srv::SetPayload::Request::SharedP
   command_interfaces_[CommandInterfaces::PAYLOAD_COG_Y].set_value(req->center_of_gravity.y);
   command_interfaces_[CommandInterfaces::PAYLOAD_COG_Z].set_value(req->center_of_gravity.z);
 
-  while (command_interfaces_[CommandInterfaces::PAYLOAD_ASYNC_SUCCESS].get_value() == ASYNC_WAITING) {
-    // Asynchronous wait until the hardware interface has set the payload
-    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+  if(!waitForAsyncCommand([&](){return command_interfaces_[CommandInterfaces::IO_ASYNC_SUCCESS].get_value();}))
+  {
+    RCLCPP_WARN(get_node()->get_logger(),"Could not verify that io was set. (This might happen when using the mocked interface)");
```

Comment:
```suggestion
    RCLCPP_WARN(get_node()->get_logger(),"Could not verify that payload was set. (This might happen when using the mocked interface)");
```